disperser/apiserver/server.go

```diff
@@ -727,6 +727,12 @@ func (s *DispersalServer) RetrieveBlob(ctx context.Context, req *pb.RetrieveBlob
 		return nil, api.NewInternalError("failed to get blob metadata, please retry")
 	}
 
+	if blobMetadata.Expiry < uint64(time.Now().Unix()) {
+		s.metrics.HandleNotFoundRpcRequest("RetrieveBlob")
+		s.metrics.HandleNotFoundRequest("RetrieveBlob")
+		return nil, api.NewNotFoundError("blob expired")
```

Comment:
should we use the same error message as L723? 
Users may be relying on the error message to handle different failure cases

Reply:
Yeah, I was wondering about this. I'll change it back for now, but we should probably revisit error message stability in the future.